Src/Tests/Temporal.Sdk.Common.Tests/Serialization/TestUnnamedContainerPayloadConverter.cs

```diff
@@ -0,0 +1,74 @@
+        }
+
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_Null()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            Payloads p = new Payloads();
+            Assert.False(instance.TrySerialize<string>(null, p));
+            Assert.Empty(p.Payloads_);
+        }
+
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_Unnamed_SerializedDataBacked()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            instance.InitDelegates(new[] { new JsonPayloadConverter() });
+            Payloads p = new Payloads();
+            NewtonsoftJsonPayloadConverter converter = new NewtonsoftJsonPayloadConverter();
+            converter.Serialize(new SerializableClass { Name = "test", Value = 2 }, p);
+            PayloadContainers.Unnamed.SerializedDataBacked data = new PayloadContainers.Unnamed.SerializedDataBacked(p, converter);
+            Assert.True(instance.TrySerialize(data, p));
+            Assert.NotEmpty(p.Payloads_);
+            Assert.True(instance.TryDeserialize(p, out PayloadContainers.Unnamed.SerializedDataBacked cl));
+            Assert.NotNull(cl);
+            SerializableClass deserializedData = cl.GetValue<SerializableClass>(0);
+            Assert.Equal("test", deserializedData.Name);
+            Assert.Equal(2, deserializedData.Value);
+        }
+
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_Unnamed_InstanceBacked()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            instance.InitDelegates(new[] { new JsonPayloadConverter() });
+            Payloads p = new Payloads();
+            PayloadContainers.Unnamed.InstanceBacked<string> data = new PayloadContainers.Unnamed.InstanceBacked<string>(new[] { "hello" });
+            Assert.True(instance.TrySerialize(data, p));
+            Assert.NotEmpty(p.Payloads_);
+            Assert.False(instance.TryDeserialize(p, out PayloadContainers.Unnamed.InstanceBacked<string> _));
+            Assert.True(instance.TryDeserialize(p, out PayloadContainers.Unnamed.SerializedDataBacked cl));
+            Assert.NotEmpty(cl);
+            Assert.True(cl.TryGetValue(0, out string val));
+            Assert.Equal("hello", val);
+        }
+
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_Unnamed_Empty()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            instance.InitDelegates(new[] { new JsonPayloadConverter() });
+            Payloads p = new Payloads();
+            PayloadContainers.Unnamed.Empty data = new PayloadContainers.Unnamed.Empty();
+            Assert.True(instance.TrySerialize(data, p));
+            Assert.Empty(p.Payloads_);
+        }
+    }
```

Comment:
I think it might be useful to also have a nested test case with a dataset along these lines:
https://github.com/temporalio/sdk-dotnet/blob/3054a2282b2eabf86d94fc970db6d4da55f8746d/Src/Demos/AdHocScenarios/Temporal.Demos.AdHocScenarios/AdHocClientInvocations.cs#L99-L108